fix(ui): update task duration in real-time when filtering by status

nitishagar opened the pull request:
When filtering tasks by status (e.g., clicking 'Running'), the Duration component was not updating in real-time. This was because react-window reuses component instances at each index, and when filtering changed which task appeared at a given index, the Duration component's interval was not properly reset.

The fix adds a key prop to the Duration (TimeDiff) component based on taskId and start time, ensuring proper component lifecycle management when the displayed task changes due to filtering or reordering.

Also adds unit tests for the Duration component covering:
- Fixed duration display (completed tasks)
- Real-time updates (running tasks)
- Key prop handling for task switching
- Interval cleanup when task completes

Fixes #8104

nitishagar commented:
@lotas / @matt-boris when you get a chance, can you review this change.

lotas commented on ui/src/components/TaskGroupTable/index.jsx:
```diff
@@ -108,7 +108,15 @@ const ItemRenderer = ({ data, index, style }) => {
-          {run ? <TimeDiff from={run.from} offset={run.to} /> : 'n/a'}
+          {run ? (
+            <TimeDiff
+              key={`${taskGroup.taskId}-${run.from}`}
```
should we also append `run.to` to the key in such case? Tasks can be running and changing states, so if it was running when first rendered, and then finished, this key would remain the same, and would likely not be updated once the task finishes.

nitishagar replied:
Spot on. The key on `TimeDiff` wouldn't help because `React.memo` prevents the parent from re-rendering - the key change would never be evaluated. Implemented using `itemKey` on the List component:
`itemKey={(index, data) => data.items[index]?.node.taskId}`

Added tests to verify the behavior.

lotas commented on ui/src/components/Duration/index.test.jsx:
```diff
@@ -0,0 +1,133 @@
+    const offset = '2025-01-01T02:30:45.000Z';
+    const { container } = render(<Duration from={from} offset={offset} />);
+
+    // Should show 2 hours 30 minutes 45 seconds
```
I feel like most of those comments are not really needed, because code below is perfectly readable, and having those extra comments don't add much besides distraction imo

lotas reviewed (changes requested):
Thanks for the PR! I feel like this should be fixed differently, because the root cause is not addressed

nitishagar commented:
Thanks for the review @lotas. The React.memo analysis was correct - the original fix addressed a symptom, not the cause. Updated to use itemKey on the List component, removed the TimeDiff key, and cleaned up test comments.

Would like your feedback on the update.
